flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/hint/LookupJoinHintOptions.java

```diff
@@ -64,17 +62,14 @@ private LookupJoinHintOptions() {}
     public static final ConfigOption<Integer> ASYNC_CAPACITY =
             key("capacity")
                     .intType()
-                    .defaultValue(
-                            ExecutionConfigOptions.TABLE_EXEC_ASYNC_LOOKUP_BUFFER_CAPACITY
-                                    .defaultValue())
+                    .noDefaultValue()
```

Comment:
for most cases, the default value could meet users' requirement, so I prefer we can give the default value

Reply:
the former version has  default value references from related ExecutionConfigOptions, but will incorrect behavior when users configured the corresponding option (differs from default one), and not explicit configured via hint, e.g.,
user set 
```
'table.exec.async-lookup.output-mode': ALLOW_UNORDERED
```
and use a LOOKUP hint
```
LOOKUP('table'='dim1', 'async'='true', 'timeout'='300s')
```
the expected final options on lookup table 'dim1' should be 
```
output-mode=ALLOW_UNORDERED, 
timeout=300s, 
buffer-capacity=100
```
but will be wrongly with default value in hint options here:
```
output-mode=ORDERED, 
timeout=300s, 
buffer-capacity=100
```

Reply:
I'll add more tests in AsyncLookupOptionsTest to cover this case.

Reply:
make sense